core/src/main/java/io/questdb/TelemetryJob.java

```diff
@@ -24,237 +24,61 @@
-        } catch (CairoException e) {
-            LOG.error()
-                    .$("Could not insert a new row in telemetry table [error=").$(e.getFlyweightMessage())
-                    .$(", errno=").$(e.getErrno())
-                    .$(']').$();
-        }
-    }
-
-    private void tryAddColumn(SqlCompiler compiler, SqlExecutionContext executionContext, CharSequence columnDetails) {
         try {
-            CompiledQuery cc = compiler.compile(
-                    "ALTER TABLE " + configTableName + " ADD COLUMN " + columnDetails,
-                    executionContext
-            );
-            try (OperationFuture fut = cc.execute(tempSequence)) {
-                fut.await();
-            }
-        } catch (SqlException ex) {
-            LOG.info().$("Failed to alter telemetry table [table=").$(configTableName).$(",error=").$(ex.getFlyweightMessage()).I$();
-        }
-    }
-
-    private TableWriter updateTelemetryConfig(
-            SqlCompiler compiler,
-            SqlExecutionContextImpl sqlExecutionContext,
-            boolean enabled,
-            TableToken tableToken
-    ) throws SqlException {
-        final TableWriter configWriter = compiler.getEngine().getWriter(AllowAllCairoSecurityContext.INSTANCE, tableToken, WRITER_LOCK_REASON);
-        final CompiledQuery cc = compiler.compile(configTableName + " LIMIT -1", sqlExecutionContext);
-        try (
-                final RecordCursorFactory factory = cc.getRecordCursorFactory();
-                final RecordCursor cursor = factory.getCursor(sqlExecutionContext)
-        ) {
-            if (cursor.hasNext()) {
-                final Record record = cursor.getRecord();
-                final boolean _enabled = record.getBool(1);
-                Long256 l256 = record.getLong256A(0);
-                final CharSequence lastVersion = record.getSym(2);
-
-                // if the configuration changed to enable or disable telemetry
-                // we need to update the table to reflect that
-                if (enabled != _enabled || !configuration.getBuildInformation().getQuestDbVersion().equals(lastVersion)) {
-                    appendConfigRow(compiler, configWriter, l256, enabled);
-                    LOG.advisory()
-                            .$("instance config changes [id=").$256(l256.getLong0(), l256.getLong1(), 0, 0)
-                            .$(", enabled=").$(enabled)
-                            .$(']').$();
-                } else {
-                    LOG.advisory()
-                            .$("instance [id=").$256(l256.getLong0(), l256.getLong1(), 0, 0)
-                            .$(", enabled=").$(enabled)
-                            .$(']').$();
-                }
-            } else {
-                // if there are no record for telemetry id we need to create one using clocks
-                appendConfigRow(compiler, configWriter, null, enabled);
-            }
+            telemetry.consumeAll();
```

Comment:
can you do try catch on each of them individually?

```java
        try {
            telemetry.consumeAll();
        } catch (Throwable th) {
            LOG.error().$("failed to process telemetry event").$(th).$();
        }

        try {
            telemetryWal.consumeAll();
        } catch (Throwable th) {
            LOG.error().$("failed to process telemetry event").$(th).$();
        }
```

Reply:
Fixed in https://github.com/questdb/questdb/pull/2930/commits/69d565260514a08087e2fd262f7576d00b5f45a5